### PR TITLE
Add filter bar to organizations page

### PR DESCRIPTION
**Pull request by miriambudayr**
Pull request for #132 

This is a work in progress. Todos:

-~~Currently we aren't filtering by location, but this is confusing because there is no way to clear location once it is set. I think if filter tags are available to click under each organization, they should also be represented in the filter bar.~~
-~Drop-down needs to expand/shrink appropriately according to text size.~
-~Fix bug where if you change the Sub Category filter to "None," the displayed text is "None" instead of the "Sub Category" placeholder text.~
-~Adjust horizontal alignment between drop-down menus and filter bar text on the left. Drop-down menus are a little too low.~
-Refactoring: I have written a few helper functions in the `OrganizationFilter.js` file, but I was wondering if we should break those out into a utils file so the component file is cleaner.

I'm happy to hear additional feedback as well. Thanks!

@jasonLaster

### CODE REVIEW

**Comment by jasonLaster** [comment]
Try running `gatsby build` locally to reproduce the build error

https://app.netlify.com/sites/sustainability-landscape/deploys/5e84c4552dbc7d0006059c1a

**Comment by jasonLaster** [comment]
Hmm, looks like we just have some sizing issues

Screen Shot 2020-04-03 at 11 07 47 AM
Screen Shot 2020-04-03 at 11 07 12 AM

**Comment by jasonLaster** [comment]
some small code suggestions
https://gist.github.com/772f54cf8aa3fee3be5a437b7155589f

I think the dropdown container should be wider and we probably shouldn't update the hover background color.

Also, I'm not sure about the white control background color. That's a question for juliana, but my take is that it clashes a bit.

File: src/components/OrganizationFilter.js [hunk 1]
```diff
@@ -95,55 +87,185 @@ export const useOrganizationFilterState = () => {
...
-            className="underline hover:no-underline ml-1"
-          >
-            clear
-          </button>
-        </p>
-      )}
+      {
+        <div className="text-gray-700  text-sm max-w-6xl mt-3">
+          <span className="mr-2 inline-block h-6 min-h-full">
+            <span>{organizations?.length} Companies | </span>
```
Comment (jasonLaster) [inline src/components/OrganizationFilter.js:239]: 👍

File: src/components/OrganizationFilter.js [hunk 2]
```diff
@@ -95,55 +87,185 @@ export const useOrganizationFilterState = () => {
...
-          {byCapitalType && (
-            <OrganizationCapitalType active text={byCapitalType} />
-          )}
-          {byCapitalStrategic && <OrganizationCapitalStrategic active />}
-          {byCapitalStage && (
-            <OrganizationCapitalStage active text={byCapitalStage} />
-          )}
-          {byCapitalCheckSize && (
-            <OrganizationCapitalCheckSize active text={byCapitalCheckSize} />
-          )}
```
Comment (jasonLaster) [inline src/components/OrganizationFilter.js:138]: Can check size should also be dropdowns on the capital page.

I'm not sure what to do with strategic which is a bool

File: src/components/OrganizationFilter.js [hunk 3]
```diff
@@ -95,55 +87,185 @@ export const useOrganizationFilterState = () => {
...
+    menu: provided => ({
+      ...provided,
+      width: "max-content",
+      minWidth: "100%",
+    }),
+
+    option: (provided, { isFocused }) => {
+      return {
+        ...provided,
+        backgroundColor: isFocused ? "#e2e8f0" : null,
```
Comment (jasonLaster) [inline src/components/OrganizationFilter.js:216]: I think the option line height should be a bit less...

File: src/components/OrganizationFilter.js (hunk 2 above)
Comment (miriambudayr) [reply src/components/OrganizationFilter.js:138]: Makes sense. I will work on the capital page.

**Comment by miriambudayr** [comment]
@jasonLaster 
What do you think about including a location dropdown? Right now location tags are available to click under each organization but there is no way to clear a location filter once it is set.

**Comment by bloudermilk** [comment]
This is great. I'd be happy to get this merged even if it's partially complete. I think we'd want to address these issues before shipping:

* Rename "None" option to "Any"
* Remove redundant subcategory list from sidebar
* Rename "Size" to "Headcount"
* Fix bug where options are removed from dropdown when one is already selected

I'm seeing the following things that could probably be filed as separate issues, roughly ordered by priority:

* Capital-specific filters on the capital page
* Add back the "clear" link after all the cells, if any are selected
* Filter by location
* Filter by role
* Minor design tweaks to the cell padding
* Using `pushState` and query parameters to preserve the filters in the URL for sharing

**Comment by miriambudayr** [comment]
@bloudermilk 
Thanks for the review. Sounds good! I am making fixes now and think I can also implement some of the fixes from the second section as well with my next push.

**Comment by bloudermilk** [comment]
@miriambudayr excellent! Not sure if you saw, but let's hold off on removing the subcategory from the sidebar for now. I'm thinking we might want to keep it there exclusively. Just opened a thread on #design if you have thoughts there!

**Comment by miriambudayr** [comment]
@bloudermilk 
Got it. To remain consistent with the /categories pages, the /capital pages would then also need a categories sidebar so that the Subcategory filter menu only includes subcategories rather than both top categories and subcategories. Is that something you want me to work on or save for later and open an issue?

**Comment by bloudermilk** [comment]
@miriambudayr yeah that deserves some though. I think the way we're using the sidebar now (Investment Type -> Stage) makes more sense than duplicating the organizations view (Category -> Subcategory). But you're right that it means we'd want to include some kind of category dropdown in the filter bar on the Capital section.

Since we will have different filters on the Capital section anyway, how about we allow for filtering by "Category" for that section only. Investors tend to focus on primary areas, not subcategories (though there are exceptions). So I think this would make sense from a UX perspective.

Anyway, I'm happy to leave off the filter bar from Capital until we can sort out the other options that need to go on that page (Check Size, Strategic, Stage, etc0

**Comment by bloudermilk** [comment]
Heads up I'm working on this right now, implementing some changes described above

**Comment by bloudermilk** [comment]
Hmmm I can't seem to push my changes to @miriambudayr PR branch, though I was able to with a PR from @jasonLaster yesterday. Anyone know why?

**Comment by bloudermilk** [comment]
@miriambudayr regarding refactoring, feel free to add files/functions to utils that you think we'll want to use elsewhere in the project. Otherwise I think it's fine we keep component-specific helpers in the component file.

**Comment by miriambudayr** [comment]
This looks really good! I will hold off on refactoring in that case. 

@bloudermilk

**Pull request merged**